docs/rfcs/rfc02.md

```diff
@@ -0,0 +1,89 @@
+# RFC02: Flags From Files (FFF)
+
+## Background
+
+(when we're speaking about flags here, we're also talking about flags mapped from env variables)
+
+Kubecfg's behaviour is affected by the presence of some flags.
+Flags can be either passed on the CLI or taken from env variables mapped to the flags.
+Flags can be problematic; not all flags do, let me tell you why.
+
+Some flags are part of the "interactive experience".
+They are effectively part of the language through which users asks kubecfg to do something for them.
+
+Some flags are user preferences. 
+
+But other flags just must be there or Jsonnet just doesn't evaluate correctly.
+Those flags are problematic.
+When the user needs to use them, they stray from the "kubecfg way".
+
+They are problematic because if they must be there, the user has two options:
+
+1. Remember to provide them every time they invoke the command
+2. Create a kubecfg wrapper script (or an env file) and remember to always use it.
+
+I've seen a lot of people doing (2), with a wrapper script that sets up all the import-path flags
+to their vendored dependencies, passing ext vars etc etc.
+
+Once the user does that, it effectively puts vital config information into a language that not particularly well suited
+for describing configs, namely, shell scripts or makefiles or whatnot.
+
+## Proposal
+
+Right before starting up, search for a `.kubecfgrc.jsonnet` in the current dir, and recursively in the parent directories.
```

Comment:
current dir is easy, but is this what we really want?

We seem to want to have a foolproof way to just evaluate a file anywhere and have it pick up the right "flags" file. That would require to locate the rc file relative to the jsonnet file about to be evaluated.

Unfortunately most kubecfg commands support multiple jsonnet files.
We could locate the kubecfgrc.jsonnet files once per input jsonnet file and apply the flags only for that file.

However, some of the flags affect the global command output and cannot be isolated on individual flags.

hmm....

Reply:
perhaps we could just search the `.kubecfgrc.jsonnet` file starting from each of the files to evaluate and error out if they differ?

Reply:
updated document with this proposal